quartz_solar_forecast/inverters/solarman.py

```diff
@@ -12,101 +12,63 @@
-        "startTime": start_time.strftime("%Y-%m-%d"),
-        "endTime": end_time.strftime("%Y-%m-%d"),
-        "deviceSn": device_info['deviceSn'],
-        "deviceId": device_info['deviceId']
-    }
-    
-    response = requests.post(url, headers=headers, json=payload)
-    
-    if response.status_code != 200:
-        raise Exception(f"Historical data API request failed with status code {response.status_code}")
-    
-    data = response.json()
-    param_data_list = data.get('paramDataList', [])
-    
-    if not param_data_list:
-        raise ValueError("No data found for the specified time range")
-    
-    # Process the data
-    rows = []
-    for entry in param_data_list:
-        timestamp = datetime.strptime(entry['collectTime'], "%Y-%m-%d").replace(tzinfo=timezone.utc)
-        generation_data = next((item for item in entry['dataList'] if item['key'] == 'generation'), None)
-        if generation_data:
-            energy_kwh = float(generation_data['value'])
-            rows.append({'timestamp': timestamp, 'energy_kwh': energy_kwh})
-    
-    # Create DataFrame
-    df = pd.DataFrame(rows)
-    df = df.sort_values('timestamp')
-    
-    print(df)
-    return df
-
-
-if __name__ == "__main__":
-    get_solarman_data()
\ No newline at end of file
+    all_data = []
+    
+    current_date = start_date
+    
+    while current_date <= end_date:
+        year = current_date.year
+        month = current_date.month
+        day = current_date.day
+        
+        url = f"{SOLARMAN_API_URL}/{SOLARMAN_ID}/record"
+        
+        headers = {
+            'Authorization': f'Bearer {SOLARMAN_TOKEN}',
+            'User-Agent': 'Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.9.0.7) Gecko/2009021910 Firefox/3.0.7'
+        }
+        
+        params = {
+            'year': year,
+            'month': month,
+            'day': day
+        }
+        
+        response = requests.get(url, headers=headers, params=params)
+        
+        if response.status_code != 200:
+            print(f"API request failed for {current_date} with status code {response.status_code}")
+        else:
+            data = response.json()
+            records = data.get('records', [])
+            
+            if records:
+                df = pd.DataFrame(records)
+                df['timestamp'] = pd.to_datetime(df['dateTime'], unit='s')
+                all_data.append(df)
+        
+        current_date += timedelta(days=1)
+    
+    if not all_data:
+        raise ValueError("No data found for the specified date range")
+    
+    # Concatenate all dataframes
+    full_data = pd.concat(all_data, ignore_index=True)
+    
+    # Select and rename relevant columns, and convert power to kW
+    full_data = full_data[['timestamp', 'generationPower']]
+    full_data = full_data.rename(columns={'generationPower': 'power_kw'})
+    full_data['power_kw'] = full_data['power_kw'] / 1000.0  # Convert watts to kilowatts
```

Comment:
This is confusing as the names of the columns doesn't show conversion from W to kW

Reply:
> This is confusing as the names of the columns doesn't show conversion from W to kW

made the change